canonical/assets.ubuntu.com
# Wd 11836

## Done

- Implement the search roughly based on [the figma](https://www.figma.com/design/dOz7GIHMmofSdje0l0pVFs/Asset-manager?node-id=0-1&node-type=canvas&t=sfLEwG7MKJ4N1XPu-0) (I took some creative license in some areas and based some other decisions off of conversations)
- I noticed there was a lot of repeating JS appearing, so I refactored the different parts into modules and allowed them to work across search and creation as they were basically doing the same thing with some minor caveats.

## QA

- Run this branch locally
- Test the search and all the different param combinations
- Check the create/update pages still work as intended

## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-11836
